### PR TITLE
SWATCH-2726: Prevent send usages to other azure tenants if not forbidden

### PR DESCRIPTION
**Pull request by Sgitario**
Jira issue: SWATCH-2726

## Description
According to the Azure Marketplace API documentation: https://learn.microsoft.com/en-us/partner-center/marketplace-offers/marketplace-metering-service-apis, it can return the following HTTP error codes:
- 400: invalid request 
- 403: forbidden 
- 409: conflict, the usage event has already been successfully reported.

The problem is that we're always throwing the "AzureMarketplaceRequestFailedException: AZURE_MARKETPLACE_API_REQUEST_FAILED: Hint: Check Azure credentials" exception regardless the HTTP error code and also trying other client tenant. 

This exception only makes sense after receiving a 403 forbidden. For the rest, we should log a trace for troubleshooting and stop trying to send the usage (since azure will return the same response).

## Changes
- Do not try to send usages or throw the exception if azure returns with an error code other than 403.
- Add test using wiremock to cover integration with azure
- Disable auto discovery of providers for Quarkus rest-client in aws and azure producers (the existing clients already configure the providers using either the property and/or the annotation)

## Testing
Added junit test to reproduce the issue. 
Since we need a Azure service, I think this can't be reproduced locally or in an ephemeral environment.

### CODE REVIEW

File: swatch-producer-azure/src/main/java/com/redhat/swatch/azure/service/AzureMarketplaceService.java [hunk 1]
```diff
@@ -60,10 +63,23 @@ public UsageEventOkResponse sendUsageEventToAzureMarketplace(UsageEvent usageEve
...
+              "Exception occurred during azure marketplace api request, likely expected since credentials are tried at random",
+              ex);
+        } else {
+          response =
+              new UsageEventOkResponse()
+                  .status(
+                      HTTP_STATUS_CONFLICT == status
+                          ? UsageEventStatusEnum.DUPLICATE
+                          : UsageEventStatusEnum.ERROR);
+          break;
```
Comment (Sgitario) [inline swatch-producer-azure/src/main/java/com/redhat/swatch/azure/service/AzureMarketplaceService.java:79]: @kflahert I think it makes sense to return this response when Azure returns these two errors, wdyt?

**Comment by Sgitario** [comment]
Changing the PR to draft to:
- add an integration test that uses wiremock to stub azure
- fix the exception mapper so the ApiExceptionMapper implementation is used when submitting the usages to azure instead of DefaultApiExceptionMapper.

**Review by kflahert — APPROVED** [review]

**Review by liwalker-rh — APPROVED** [review]
Testing and latest PR checks look good to me.

**Pull request merged**